### PR TITLE
added code for encodeProtobuf

### PR DESCRIPTION
**Pull request by NeerajGartia21**

### CODE REVIEW

File: lib/protocols.js [hunk 1]
```diff
@@ -1001,6 +1025,21 @@
     }
   }());
 
+  var WrappedMessage = (function() {
+    var root=protobuf.loadSync(path.join(__dirname+'/protostream/message-wrapping.proto'));
```
Comment (rigazilla) [inline lib/protocols.js:1029]: Will the `message-wrapping.proto` file be shipped with the library? I mean are we sure that the `loadSync` will find it the lib will be installed on the user side?

Comment (NeerajGartia21) [reply lib/protocols.js:1029]: I think it will because the path of the file is not relative. Also I have tested it.

**Comment by rigazilla** [comment]
@NeerajGartia21 I've added some more commits:
- some fixes for hotrod encode bytes: length must be sent before the buffer
- fix in WrappedMessage field population
- added protostream encoded cache and tests

Hey! encoding seems to work very well! Awesome!

if it looks good to you I can merge all this into main

**Comment by NeerajGartia21** [comment]
It looks fine to me.

**Comment by rigazilla** [comment]
merged, thanks @NeerajGartia21 !

**Comment by NeerajGartia21** [comment]
Thanks @rigazilla .

**Pull request merged**